pkg/dockerregistry/server/auth.go

```diff
@@ -167,16 +153,57 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 					challenge.err = err
 					return nil, challenge
 				}
-
-				return ctx, nil
```

Comment:
removed in lieu of the return at the bottom of the function. should we have been returning `WithUserClient(ctx, client)` here?